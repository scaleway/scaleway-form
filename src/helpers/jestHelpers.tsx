import { ThemeProvider } from '@emotion/react'
import makeHelpers from '@scaleway/jest-helpers'
import { theme as scwuiTheme } from '@scaleway/ui'
import { RenderResult, render } from '@testing-library/react'
import React, { FC, ReactElement, ReactNode } from 'react'
import Form from '../components/Form'
import mockErrors from '../mocks/mockErrors'

interface WrapperProps {
  theme?: typeof scwuiTheme
}

const Wrapper: FC<WrapperProps> = ({ theme = scwuiTheme, children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const {
  renderWithTheme,
  shouldMatchEmotionSnapshot,
  shouldMatchEmotionSnapshotWithPortal,
} = makeHelpers(Wrapper, { classNameReplacer: className => className })

export const shouldMatchEmotionSnapshotFormWrapper = (
  children: ReactNode,
  options?: {
    transform: (node: RenderResult) => Promise<void> | void
  },
) =>
  shouldMatchEmotionSnapshot(
    <Form errors={mockErrors}>{() => children}</Form>,
    options,
  )

export const renderWithWrapper = (children: ReactElement) =>
  render(children, {
    wrapper: () => (
      <ThemeProvider theme={scwuiTheme}>
        <Form errors={mockErrors}>{children}</Form>
      </ThemeProvider>
    ),
  })

export const mockRandom = () =>
  jest.spyOn(global.Math, 'random').mockReturnValue(0.4155913669444804)

export const restoreRandom = () =>
  jest.spyOn(global.Math, 'random').mockRestore()
