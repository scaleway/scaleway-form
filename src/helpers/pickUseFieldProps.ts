import { AnyObject } from 'final-form'
import { UseFieldConfig } from 'react-final-form'
import { BaseFieldProps } from '../types'

const pickUseFieldProps = <T = AnyObject, K = AnyObject>({
  afterSubmit,
  allowNull,
  beforeSubmit,
  data,
  defaultValue,
  format,
  formatOnBlur,
  initialValue,
  isEqual,
  multiple,
  parse,
  subscription,
  validate,
  validateFields,
  value,
}: BaseFieldProps<T, K>): UseFieldConfig<T, K> => ({
  afterSubmit,
  allowNull,
  beforeSubmit,
  data,
  defaultValue,
  format,
  formatOnBlur,
  initialValue,
  isEqual,
  multiple,
  parse,
  subscription,
  validate,
  validateFields,
  value,
})

export default pickUseFieldProps
