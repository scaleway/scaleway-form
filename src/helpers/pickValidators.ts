import type { ValidatorObject, ValidatorProps } from '../types'
import validators from '../validators'
import type { ValidatorFn } from '../validators/types'

export const pickValidators = <InputValue = unknown>(args: ValidatorProps) =>
  Object.entries(args)
    .map(([key, value]) =>
      value !== undefined
        ? (validators as Record<string, ValidatorFn>)[
            key as keyof typeof validators
          ]?.(value)
        : undefined,
    )
    .filter(validator => !!validator) as ValidatorObject<InputValue>[]
