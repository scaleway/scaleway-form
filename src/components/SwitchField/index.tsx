import { Box, Switch, Typography } from '@scaleway/ui'
import React, { ReactNode } from 'react'
import { useField } from 'react-final-form'
import pickUseFieldProps from '../../helpers/pickUseFieldProps'
import pickValidators from '../../helpers/pickValidators'
import useValidation from '../../hooks/useValidation'
import { BaseFieldProps } from '../../types'

export type SwitchFieldProps<T = unknown, K = unknown> = BaseFieldProps<
  T,
  K
> & {
  required?: boolean
  label?: string
  variant?: 'primary' | 'success'
  size?: 'small' | 'medium'
  tooltip?: string
  width?: number
  labeled?: 'inside' | 'left' | 'right' | boolean
  offLabel?: ReactNode
  onLabel?: ReactNode
  name: string
  disabled?: boolean
}
const SwitchField = ({
  label,
  variant,
  size = 'small',
  tooltip,
  width,
  onLabel = 'ON',
  offLabel = 'OFF',
  labeled,
  name,
  defaultValue,
  disabled,
  validate,
  required,
  ...props
}: SwitchFieldProps) => {
  const validateFn = useValidation(pickValidators({ required }))

  const { input } = useField(name, {
    ...pickUseFieldProps(props),
    type: 'checkbox',
    validate: validateFn,
  })

  return (
    <Box display="flex" alignItems="center">
      <Switch
        checked={input.checked}
        variant={variant}
        width={width}
        tooltip={tooltip}
        onChange={input.onChange}
        onLabel={onLabel}
        offLabel={offLabel}
        labeled={labeled}
        size={size}
        name={name}
        disabled={disabled}
      />
      {label && (
        <Typography ml={1} color={input.checked ? 'primary' : 'gray950'}>
          {label}
        </Typography>
      )}
    </Box>
  )
}

export default SwitchField
