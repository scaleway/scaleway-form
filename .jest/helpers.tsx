import { ThemeProvider } from '@emotion/react'
import makeHelpers from '@scaleway/jest-helpers'
import { theme as lightTheme } from '@scaleway/ui'
import { render } from '@testing-library/react'
import type { ComponentProps, ReactElement, ReactNode } from 'react'
import { Form } from '../src/components'
import { mockErrors } from '../src/mocks'

type WrapperProps = {
  theme?: typeof lightTheme
  children: ReactNode
}

const Wrapper = ({ theme = lightTheme, children }: WrapperProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const {
  renderWithTheme,
  shouldMatchEmotionSnapshot,
  shouldMatchEmotionSnapshotWithPortal,
} = makeHelpers(Wrapper, { classNameReplacer: className => className })

export const shouldMatchEmotionSnapshotFormWrapper = (
  children: ReactNode,
  options?: Parameters<typeof shouldMatchEmotionSnapshot>[1],
  formOptions?: Partial<ComponentProps<typeof Form>>,
) => {
  const { initialValues } = formOptions ?? {}

  return shouldMatchEmotionSnapshot(
    <Form
      onRawSubmit={() => {}}
      errors={mockErrors}
      initialValues={initialValues}
    >
      {() => children}
    </Form>,
    options,
  )
}

export const renderWithWrapper = (children: ReactElement) =>
  render(children, {
    wrapper: () => (
      <ThemeProvider theme={lightTheme}>
        <Form onRawSubmit={() => {}} errors={mockErrors}>
          {children}
        </Form>
      </ThemeProvider>
    ),
  })

export const mockRandom = () =>
  jest.spyOn(global.Math, 'random').mockReturnValue(0.4155913669444804)

export const restoreRandom = () =>
  jest.spyOn(global.Math, 'random').mockRestore()
