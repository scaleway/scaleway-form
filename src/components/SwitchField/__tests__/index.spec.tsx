import React from 'react'
import SwitchField from '..'
import { shouldMatchEmotionSnapshotFormWrapper } from '../../../helpers/jestHelpers'

describe('SwitchField', () => {
  test('should render correctly', () =>
    shouldMatchEmotionSnapshotFormWrapper(<SwitchField name="test" />))

  test('should render correctly disabled', () =>
    shouldMatchEmotionSnapshotFormWrapper(<SwitchField name="test" disabled />))

  test('should render correctly checked', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SwitchField name="test" value="test" initialValue={['test']} />,
      {
        transform: node => {
          const element = node.getByRole('checkbox') as HTMLInputElement
          expect(element.checked).toBeTruthy()
        },
      },
    ))
  test('should render correctly with onLabel', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SwitchField name="test" onLabel="Checked" />,
    ))
  test('should render correctly with offLabel', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SwitchField name="test" onLabel="Not checked" />,
    ))

  test('should render correctly without onLabel and offLabel but has label', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SwitchField name="test" onLabel="" offLabel="" label="test" />,
    ))

  test('should render correctly with label and checked', () =>
    shouldMatchEmotionSnapshotFormWrapper(
      <SwitchField name="test" initialValue label="test" />,
    ))
})
