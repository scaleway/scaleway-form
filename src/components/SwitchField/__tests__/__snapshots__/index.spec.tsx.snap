// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchField should render correctly 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="false"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="false"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="false"
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly checked 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="true"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="true"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="true"
            checked=""
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly disabled 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-l72xnd-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.cache-1etx9wf-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #dce1eb;
  border: none;
  border-radius: 34px;
  cursor: not-allowed;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 0.5;
}

.cache-1etx9wf-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1etx9wf-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1etx9wf-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1etx9wf-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="false"
        class="cache-l72xnd-StyledLabel e1x48wse0"
        disabled=""
      >
        <span
          aria-checked="false"
          class="cache-1etx9wf-StyledSwitch-buildVariants e1x48wse2"
          disabled=""
        >
          <input
            aria-checked="false"
            aria-disabled="true"
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            disabled=""
            id="test"
            name="test"
            style="pointer-events: none;"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly with label and checked 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

.cache-16o8nhl-StyledBox-StyledText-variantStyles-colorStyles {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4f0599;
}

.cache-16o8nhl-StyledBox-StyledText-variantStyles-colorStyles.cache-16o8nhl-StyledBox-StyledText-variantStyles-colorStyles {
  color: #4f0599;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="true"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="true"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="true"
            checked=""
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
      <div
        class="e1g1zfwd0 cache-16o8nhl-StyledBox-StyledText-variantStyles-colorStyles e7ah3bn0"
      >
        test
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly with offLabel 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="false"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="false"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="false"
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly with onLabel 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="false"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="false"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="false"
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SwitchField should render correctly without onLabel and offLabel but has label 1`] = `
<DocumentFragment>
  .cache-axw7ok {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.cache-1oshk27-StyledLabel {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.cache-1axmni8-StyledSwitch-buildVariants {
  overflow: hidden;
  outline: none;
  width: 40px;
  height: 24px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b2b6c3;
  border: none;
  border-radius: 34px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  padding: 0 6px;
  opacity: 1;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true'] {
  background-color: #4f0599;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse4 {
  -webkit-transform: translateX(
              28px
            ) translateX(-100%);
  -moz-transform: translateX(
              28px
            ) translateX(-100%);
  -ms-transform: translateX(
              28px
            ) translateX(-100%);
  transform: translateX(
              28px
            ) translateX(-100%);
  background-color: #f6f5f7;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='false']>.e1x48wse3 {
  color: #4a4f62;
  right: 0;
}

.cache-1axmni8-StyledSwitch-buildVariants[aria-checked='true']>.e1x48wse3 {
  color: #f6f5f7;
  left: 0;
}

.cache-16qhhwn-StyledCheckbox {
  position: absolute;
  opacity: 0.01;
}

.cache-1kh0bkm-SwitchBall {
  display: block;
  position: relative;
  border-radius: 34px;
  pointer-events: none;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  width: 12px;
  height: 12px;
  background-color: #f6f5f7;
}

.cache-1f81ejt-StyledBox-StyledText-variantStyles-colorStyles {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #151a2d;
}

.cache-1f81ejt-StyledBox-StyledText-variantStyles-colorStyles.cache-1f81ejt-StyledBox-StyledText-variantStyles-colorStyles {
  color: #151a2d;
}

<form
    novalidate=""
  >
    <div
      class="cache-axw7ok e1fcp5360"
    >
      <label
        aria-checked="false"
        class="cache-1oshk27-StyledLabel e1x48wse0"
      >
        <span
          aria-checked="false"
          class="cache-1axmni8-StyledSwitch-buildVariants e1x48wse2"
        >
          <input
            aria-checked="false"
            class="cache-16qhhwn-StyledCheckbox e1x48wse1"
            id="test"
            name="test"
            type="checkbox"
          />
          <span
            class="cache-1kh0bkm-SwitchBall e1x48wse4"
          />
        </span>
      </label>
      <div
        class="e1g1zfwd0 cache-1f81ejt-StyledBox-StyledText-variantStyles-colorStyles e7ah3bn0"
      >
        test
      </div>
    </div>
  </form>
</DocumentFragment>
`;
